flowey/schema_ado_yaml/src/lib.rs

```diff
@@ -87,7 +97,10 @@ pub enum CiTrigger {
     #[serde(rename_all = "camelCase")]
     Some {
         batch: bool,
-        branches: TriggerBranches,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        branches: Option<TriggerBranches>,
```

Comment:
The issue here was if branches was empty, the generated yaml would get an entry that looked like
```
  tags: {}
```

Reply:
Might be worth investigating if that's an issue with the yaml crate being used. Sounds like a upstream bug

Reply:
I'm not sure this is an issue with the yaml crate, seems like this is a valid fix to omit empty elements.